jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -162,14 +162,23 @@ public Response createEntitlement(final SubscriptionJson entitlement,
                              entitlement.getPriceList(), "SubscriptionJson priceList needs to be set");
 
         final Iterable<PluginProperty> pluginProperties = extractPluginProperties(pluginPropertiesString);
+        final CallContext callContext = context.createContext(createdBy, reason, comment, request);
+
+        SubscriptionBundle bundle = null;
         final boolean createAddOnEntitlement = ProductCategory.ADD_ON.toString().equals(entitlement.getProductCategory());
         if (createAddOnEntitlement) {
-            verifyNonNullOrEmpty(entitlement.getBundleId(), "SubscriptionJson bundleId should be specified for ADD_ON");
+            verifyNonNullOrEmpty(entitlement.getExternalKey(), "SubscriptionJson externalKey should be specified for ADD_ON");
+            try {
```

Comment:
You should allow either a `bundleId` or `bundleExternalKey`. If both are specified the code would use the `bundleId` and ignore the `bundleExternalKey` (ideally we should check the two point to the same object and throw an error if this is not the case, but i am sure there are many places where we don't check, so let's keep it simple).